ament_lint_auto/cmake/ament_lint_auto_find_test_dependencies.cmake

```diff
@@ -0,0 +1,26 @@
+#
+# Invoke find_package() for all test dependencies.
+#
+# All found package names are appended to the
+# ``${PROJECT_NAME}_FOUND_TEST_DEPENDS`` variables.
+#
+# @public
+#
+macro(ament_lint_auto_find_test_dependencies)
+  if(ARGN)
+    message(FATAL_ERROR "ament_lint_auto_find_test_dependencies() called with "
+      "unused arguments: ${ARGN}")
+  endif()
+
+  if(NOT _AMENT_PACKAGE_NAME)
+    ament_package_xml()
+  endif()
+
+  # try to find_package() all test dependencies
+  foreach(_dep ${${PROJECT_NAME}_TEST_DEPENDS})
+    find_package(${_dep} QUIET)
+    if(${_dep}_FOUND)
+      list(APPEND ${PROJECT_NAME}_FOUND_TEST_DEPENDS ${_dep})
+    endif()
+  endforeach()
+endmacro()
```

Comment:
In the example the code block must be wrapped in a check for `AMENT_ENABLE_TESTING` anyway since otherwise `ament_lint_auto` is not even `find_package()`-able.

Also all packages registering tests are designed to work fine even is testing is disabled. Since all test targets are managed by CMake they will just be skipped. So no need to wrap all tests in `AMENT_ENABLE_TESTING` blocks anymore.